components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/java/org/wso2/carbon/identity/api/server/tenant/management/v1/impl/TenantsApiServiceImpl.java

```diff
@@ -74,4 +74,20 @@ private URI getResourceLocation(String resourceId) {
         return ContextLoader.buildURIForHeader(Constants.V1_API_PATH_COMPONENT +
                 TenantManagementConstants.TENANT_MANAGEMENT_PATH_COMPONENT + "/" + resourceId);
     }
+
+    @Override
+    public Response getTenantByDomain(String tenantDomain) {
+
+        return Response.ok().entity(tenantManagementService.getTenantByDomain(tenantDomain)).build();
+    }
+
+    @Override
+    public Response isDomainAvailable(String tenantDomain) {
```

Comment:
The method name and the response is contracting, isn't it? 

we return NOT_FOUND if the domain is available, 

and

Return OK if the domain is not available

Reply:
That has been the usual practice I saw in domain checking..   In google, if the email is available for us to be taken and not found in the system, they say it is available. Same with DNS domain registries. Here available implies it is available for sale or to be taken kind of.. WDYT?

Reply:
Maybe we can change it to: isDomainExists?

Reply:
Addressed.